x-pack/plugins/security_solution/public/common/components/endpoint/responder/from_alerts/use_responder_action_data.test.ts

```diff
@@ -0,0 +1,283 @@
+
+  const getExpectedResponderActionData = (
+    overrides: Partial<ResponderActionData> = {}
+  ): ResponderActionData => {
+    return {
+      isDisabled: false,
+      tooltip: undefined,
+      handleResponseActionsClick: expect.any(Function),
+      ...overrides,
+    };
+  };
+
+  beforeEach(() => {
+    appContextMock = createAppRootMockRenderer();
+    onClickMock = jest.fn();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('useWithResponderActionDataFromAlert() hook', () => {
+    let renderHook: () => RenderHookResult<
+      UseWithResponderActionDataFromAlertProps,
+      ResponderActionData
+    >;
+    let alertDetailItemData: TimelineEventsDetailsItem[];
+
+    beforeEach(() => {
+      renderHook = () => {
+        return appContextMock.renderHook<
+          UseWithResponderActionDataFromAlertProps,
+          ResponderActionData
+        >(() =>
+          useWithResponderActionDataFromAlert({
+            eventData: alertDetailItemData,
+            onClick: onClickMock,
+          })
+        );
+      };
+    });
+
+    describe('Common behaviours', () => {
+      it('should show action as disabled if agent does not support response actions', () => {
+        alertDetailItemData = endpointAlertDataMock.generateAlertDetailsItemDataForAgentType('foo');
+
+        expect(renderHook().result.current).toEqual(
+          getExpectedResponderActionData({
+            isDisabled: true,
+            tooltip: NOT_FROM_ENDPOINT_HOST_TOOLTIP,
+          })
+        );
+      });
+
+      it('should call `onClick()` function prop when is pass to the hook', () => {
+        alertDetailItemData = endpointAlertDataMock.generateSentinelOneAlertDetailsItemData();
+        const { result } = renderHook();
+        result.current.handleResponseActionsClick();
+
+        expect(onClickMock).toHaveBeenCalled();
+      });
+
+      it('should NOT call `onClick` if the action is disabled', () => {
+        alertDetailItemData = endpointAlertDataMock.generateAlertDetailsItemDataForAgentType('foo');
+        const { result } = renderHook();
+        result.current.handleResponseActionsClick();
+
+        expect(onClickMock).not.toHaveBeenCalled();
+      });
+    });
+
+    describe('and agentType is NOT Endpoint', () => {
+      beforeEach(() => {
+        alertDetailItemData = endpointAlertDataMock.generateSentinelOneAlertDetailsItemData();
+      });
+
+      it('should show action when agentType is supported', () => {
+        expect(renderHook().result.current).toEqual(getExpectedResponderActionData());
+      });
+
+      it('should NOT call the endpoint host metadata api', () => {
+        renderHook();
+        const wasMetadataApiCalled = appContextMock.coreStart.http.get.mock.calls.some(([path]) => {
+          return (path as unknown as string).includes(HOST_METADATA_LIST_ROUTE);
+        });
+
+        expect(wasMetadataApiCalled).toBe(false);
+      });
+
+      it.each([...RESPONSE_ACTION_AGENT_TYPE])(
```

Comment:
This test also includes the Endpoint agent

Reply:
I notice that the enclosing describe says the tests are for non endpoint agents so I presume these tests should be only for external agents.

Reply:
ahhh. Ok. let me see if maybe I placed the test in the wrong place

Reply:
Ok. thats what it was. Nice catch. I moved that test (`it.each`) to the "Common behaviors" `describe()` block